libmamba/include/mamba/solver/solution.hpp

```diff
@@ -67,22 +67,23 @@ namespace mamba::solver
         using action_list = std::vector<Action>;
 
         action_list actions = {};
-    };
 
-    template <typename Iter, typename UnaryFunc>
-    void for_each_to_remove(Iter first, Iter last, UnaryFunc&& func);
-    template <typename Range, typename UnaryFunc>
-    void for_each_to_remove(Range&& actions, UnaryFunc&& func);
+        /** Return a view of all unique packages involved in the solution. */
```

Comment:
I think it would be helpful to clarify in these documentation the validity of these views when modifying member functions are called. Are they invalid? Depends on what is called? When it's called? etc. (kind of same way we specify the validity of iterators for a container)